ui/ts/pages/stores.ts

```diff
@@ -138,7 +141,24 @@ module AdminViews {
      * StorePage show the details of a single node.
      */
     export module StorePage {
+      import NavigationBar = Components.NavigationBar;
+
       class Controller {
+        private static defaultTargets: NavigationBar.Target[] = [
+          {
+            title: "Overview",
+            route: ""
+          },
+          {
+            title: "Graphs",
+            route: "graph"
+          }
+        ];
+
+        private static isActive: (targ: NavigationBar.Target) => boolean = (t: NavigationBar.Target) => {
```

Comment:
Instead of one function that returns a target, how about doing the opposite, have a function on each target, that determines if it is the active one?
This way, you could make it a readOnlyProp that depends on the route.